File: framework/src/main/java/com/framework/utils/MatcherUtils.java
package com.framework.utils;

import java.util.regex.Pattern;

/**
 * by y on 2017/4/6.
 */

public class MatcherUtils {

    private MatcherUtils() {
    }

    /**
     * 获取String里面的所有数字
     */
    public static int getInt(String s) {
        return Integer.valueOf(Pattern.compile("[^0-9]").matcher(s).replaceAll("").trim());
    }

    /**
     * 获取妹子图搜索结果的总页数
     */
    public static Integer getIntHasSpace(String s) {
        String replace = s.replace("下一页»", "").replace("«上一页", "");
        String[] split = Pattern.compile("[^0-9]").matcher(replace).replaceAll("/").split("/");
        if (split.length >= 1) {
            return Integer.valueOf(split[split.length - 1]);
        } else {
            return Integer.valueOf(split[0]);
        }
    }

}

File: framework/src/main/java/com/framework/utils/MatcherUtils.kt
package com.framework.utils

import java.util.regex.Pattern

/**
 * by y on 2017/4/6.
 */

object MatcherUtils {

    /**
     * 获取String里面的所有数字
     */
    fun getInt(s: String): Int {
        return Integer.valueOf(Pattern.compile("[^0-9]").matcher(s).replaceAll("").trim { it <= ' ' })!!
    }

    /**
     * 获取妹子图搜索结果的总页数
     */
    fun getIntHasSpace(s: String): Int? {
        val replace = s.replace("下一页»", "").replace("«上一页", "")
        val split = Pattern.compile("[^0-9]").matcher(replace).replaceAll("/").split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        if (split.isNotEmpty()) {
            return Integer.valueOf(split[split.size - 1])
        } else {
            return Integer.valueOf(split[0])
        }
    }

}
